plugins/woocommerce-admin/client/customer-effort-score-tracks/customer-effort-score-exit-page.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { dispatch, resolveSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ALLOW_TRACKING_OPTION_NAME } from './constants';
+
+const CUSTOMER_EFFORT_SCORE_EXIT_PAGE_KEY = 'customer-effort-score-exit-page';
+
+let allowTracking = false;
+resolveSelect( OPTIONS_STORE_NAME )
+	.getOption( ALLOW_TRACKING_OPTION_NAME )
+	.then( ( trackingOption ) => {
+		allowTracking = trackingOption === 'yes';
+	} );
+
+/**
+ * Gets the list of exited pages from Localstorage.
+ */
+export const getExitPageData = () => {
+	if ( ! window.localStorage ) {
+		return [];
+	}
+
+	const items = window.localStorage.getItem(
+		CUSTOMER_EFFORT_SCORE_EXIT_PAGE_KEY
+	);
+	const parsedJSONItems = items ? JSON.parse( items ) : [];
+	const arrayItems = Array.isArray( parsedJSONItems ) ? parsedJSONItems : [];
+
+	return arrayItems;
+};
+
+/**
+ * Adds the page to the exit page list in Localstorage.
+ *
+ * @param {string} pageId of page exited early.
+ */
+export const addExitPage = ( pageId: string ) => {
+	if ( ! window.localStorage ) {
+		return;
+	}
+
+	let items = getExitPageData();
+
+	items = items.filter( ( item ) => item !== pageId );
```

Comment:
I'm not clear about this part. Why do we remove the item from the array and add it again?
Maybe we can check if the item is present in the array and add it if not.
It would look like this:
```
if ( ! items.find( ( pageExitedId ) => pageExitedId === pageId ) ) {
    items.push( pageId );
}
```

Reply:
Simplicity mostly, but I can use your approach as well.

Reply:
Also to make sure we never have duplicates.